apps/payments/e2e/objects/paymentRequests/PaymentRequestDetailsPage.ts

```
@@ -0,0 +1,148 @@
+    await expect(this.page.getByText(description)).toBeVisible();
+  }
+
+  async checkStatus(status: string) {
+    await expect(this.statusLabel).toBeVisible();
+    await expect(this.status(status)).toBeVisible();
+  }
+
+  async checkAccounts(
+    providers: {
+      name: string;
+      type: ProviderType;
+    }[],
+  ) {
+    await Promise.all(
+      providers.map(
+        async (provider) =>
+          await this.page
+            .locator("div")
+            .filter({ hasText: providerTypeAccountLabelMap[provider.type] })
+            .last()
+            .getByText(provider.name),
+      ),
+    );
+  }
+
+  async checkAmount(amount: string) {
+    await expect(this.amountLabel).toBeVisible();
+    await expect(this.page.getByText(amount)).toBeVisible();
+  }
+
+  async checkRedirectUrl(url: string) {
+    await expect(this.redirectURLLabel).toBeVisible();
+    await expect(this.page.getByText(url)).toBeVisible();
+  }
+
+  async checkAmountOverrideOption(bool: boolean) {
+    await expect(this.amountOverrideLabel).toBeVisible();
+    await expect(this.allowAmountOverride(bool.toString())).toBeVisible();
+  }
+
+  async checkCustomAmountOption(bool: boolean) {
+    await expect(this.customAmountLabel).toBeVisible();
+    await expect(this.allowCustomAmount(bool.toString())).toBeVisible();
+  }
+
+  async checkEmptyPaymentsList() {
+    await expect(
+      this.page.getByRole("heading", { name: "Payments Received" }),
+    ).toBeVisible();
+    await expect(
+      this.page.getByRole("heading", { name: "There are no payments yet" }),
+    ).toBeVisible();
+    await expect(
+      this.page.getByText(
+        "You have not received any payments yet for this request",
+      ),
+    ).toBeVisible();
+  }
+
+  async delete() {
+    await this.deleteButton.click();
+  }
+
+  async getPaymentLink() {
+    await expect(this.paymentLinkLabel).toBeVisible();
+    return (await this.paymentLink.textContent()) ?? "";
```

Is `?? ""` really necessary? Shall we expect that `paymentLink.textContent()` is not empty and then cast as string if necessary? I'd prefer throwing an error rather than returning an empty string if the payment link is not generated correcly

---

We expect the payment link to always be defined, but `locator.textContent()` returns string or undefined by signature. There is usually no need to throw custom error in e2e tests as assertions already cover that. If the link is empty the test will fail at the following assertion, when trying to open a page from an empty link :)